### DISCUSSION TITLE
fastai training style with Neural Prophet

### DISCUSSION THREAD

**Discussion by turgut090** [ourownstory/neural_prophet#210] [Feature Requests & Feedback]
Hi, Is there  a way to train the neural prophet model like we do in fast.ai?

Like in this notebook but for Neural Prophet: https://github.com/ourownstory/AR-Net/blob/master/example_notebooks/arnet.ipynb

E.g. find lr and get similar output:
```
epoch	train_loss	valid_loss	mae	sTPE of AR coeff	time
0	0.031085	0.021893	0.119834	54.384563	00:01
1	0.013529	0.010846	0.083358	18.060856	00:01
2	0.012378	0.010164	0.080609	5.892164	00:01
3	0.012762	0.010338	0.081312	9.736054	00:01
4	0.013084	0.010232	0.080985	7.716311	00:01
5	0.012711	0.010290	0.081196	10.196399	00:01
6	0.012592	0.010251	0.081023	7.288769	00:01
7	0.012658	0.010218	0.080901	5.527937	00:01
8	0.012589	0.010339	0.081309	12.008403	00:01
9	0.012473	0.010184	0.080733	6.603688	00:01
10	0.012482	0.010169	0.080695	6.209393	00:01
11	0.012320	0.010151	0.080549	4.760333	00:01
12	0.011984	0.010150	0.080568	4.440498	00:01
13	0.011870	0.010148	0.080574	4.290320	00:01
14	0.011768	0.010149	0.080570	4.222988	00:01
```

**Comment by ourownstory** [comment]
@henry090 We did consider using fastai2 for this project, but decided not to, as we take a lot of low-level control of the model and training in PyTorch. Further, we tried to retain the UI of fbprophet as far as possible.

However, you can get similar behavior.
* if you do not specify lr, a lr range test is done.
* check out `validate_each_epoch` in `fit`
* check out `plot_live_loss` in `fit`
* check out the metrics_df returned by `fit`

If you have specific suggestions, please let us know!
